### PR TITLE
registrar: parse EK cert with pyasn1

### PR DESCRIPTION
**Pull request by sergio-correia**
python-cryptography won't parse malformed EK certs, such as some
Nuvoton ones we have encountered in the field. However, we still
have to deal with these certs anyway, so let's use pyasn1 to parse
the EK cert instead.

### CODE REVIEW

**Comment by sergio-correia** [comment]
@mpeters @THS-on @maugustosilva @aplanas @kkaarreell: this works around the issue in #944. Please let me know your thoughts on an approach like this.

**Comment by THS-on** [comment]
I think it is the approach is fine and pyasn1 is packaged for most distributions. The only thing that worries me slightly is that pyasn1 had its last release 2019. Is this parser used by any other major projects?

**Comment by mpeters** [comment]
I think it's a good approach. The fact that it hasn't been updated is a bit concerning, but if it's doing just cert parsing and the specs are pretty solid, maybe it hasn't had much to update. Are there any outstanding PRs? What is it's history of CVEs?

As for the testing-farm failures, maybe they need to be updated to install the deps via package managers?

**Comment by kkaarreell** [comment]
Could you please create PR extending /setup/install_upstream_keylime with additional dependencies? You can also temporarily point packit-ci.fmf to your fork and branch to test it.

**Comment by sergio-correia** [comment]
> I think it is the approach is fine and pyasn1 is packaged for most distributions. The only thing that worries me slightly is that pyasn1 had its last release 2019. Is this parser used by any other major projects?

Some projects using it:
- freeipa: https://pagure.io/freeipa
- 389 directory server: https://github.com/389ds/389-ds-base.git
- google auth python: https://github.com/googleapis/google-auth-library-python
- python-ldap: https://github.com/python-ldap/python-ldap
- tpm2-pkcs11: https://github.com/tpm2-software/tpm2-pkcs11

**Comment by sergio-correia** [comment]
> I think it's a good approach. The fact that it hasn't been updated is a bit concerning, but if it's doing just cert parsing and the specs are pretty solid, maybe it hasn't had much to update. Are there any outstanding PRs? What is it's history of CVEs?
> 

There are a few PRs: https://github.com/etingof/pyasn1/pulls

I couldn't find much in terms of CVE history; there are a few hits when you do a broad search using it as a keyword, but I would have to dig deeper to have a better picture here.

> As for the testing-farm failures, maybe they need to be updated to install the deps via package managers?

Yep, I will follow Karel's suggestion and update `packit-ci.fmf` temporarily

File: keylime/cert_utils.py
Comment (aplanas) [inline keylime/cert_utils.py:11]: nit: no need to add the name of the function in the doc string

File: packit-ci.fmf
Comment (aplanas) [inline packit-ci.fmf:16]: It is referencing your private github. It did slip in the commit? If not, maybe is better to add a new PR in the RedHat repo and wait before merging this.

**Comment by aplanas** [comment]
Some clarifications:
* Ownership of pyasn1 has been transferred as the maintainer dropped them: https://github.com/etingof/pyasn1/issues/214
* The new project is living in https://github.com/pyasn1/
* Is has been tagged for 0.5.0 RC2 just some hrs ago

So I think that we can use the module safely.

**Review by aplanas — APPROVED** [review]

File: keylime/cert_utils.py
Comment (sergio-correia) [reply keylime/cert_utils.py]: Thanks, resolved.

File: packit-ci.fmf
Comment (sergio-correia) [reply packit-ci.fmf]: Yeah, it was a temp change to run the tests with some different setup (my fork installed `pyasn1`). I removed this change now.

**Comment by mpeters** [comment]
@sergio-correia are we ready to merge this?

**Comment by sergio-correia** [comment]
> @sergio-correia are we ready to merge this?

Yep, feel free to go ahead.

**Review by mpeters — APPROVED** [review]

**Pull request merged**
